storage/checkpoint.rs

```diff
@@ -0,0 +1,292 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use std::{
+    collections::BTreeMap,
+    error::Error,
+    fmt,
+    fs::{self, File},
+    io::{self, Write},
+    path::{Path, PathBuf},
+};
+
+use chrono::Utc;
+use durability::{DurabilityError, DurabilityRecord, DurabilityService, RawRecord, SequenceNumber};
+use itertools::Itertools;
+
+use crate::{
+    isolation_manager::{CommitRecord, IsolationManager, StatusRecord, ValidatedCommit},
+    keyspace::{
+        KeyspaceCheckpointError, KeyspaceError, KeyspaceOpenError, KeyspaceSet, KeyspaceValidationError, Keyspaces,
+    },
+    write_batches::WriteBatches,
+    MVCCStorage, StorageCommitError,
+};
+
+pub(crate) struct Checkpoint {
+    pub keyspaces: Keyspaces,
+    pub next_sequence_number: SequenceNumber,
+    pub directory: Option<PathBuf>,
+}
```

Comment:
Doesn't really make sense to call this struct a `Checkpoint`, this is more a result of restoring _from_ a checkpoint (and a WAL)